logstash-core/lib/logstash/execution_context.rb

```diff
@@ -0,0 +1,10 @@
+# encoding: utf-8
+module LogStash
+  class ExecutionContext
+    attr_reader :pipeline_id
```

Comment:
@talevy the `path.data` is already available in the `LogStash::SETTINGS`, I don't we need to pass it using the execution context. The value should be valid and never change for the process.